bin/keycloak-httpd-client-install

```diff
@@ -1218,39 +1289,54 @@ def main():
                                           loader=jinja2.FileSystemLoader(
-                                              options.template_dir))
+                                              'templates'))
     except Exception as e:
         if options.show_traceback:
             traceback.print_exc()
         print('%s: %s' % (e.__class__.__name__, six.text_type(e)),
               file=sys.stderr)
         return STATUS_CONFIGURATION_ERROR
 
-    # ===== Congfigure Mellon  =====
+    if options.module == 'mod_auth_mellon':
+        # ===== Congfigure Mellon  =====
 
-    try:
-        logger.step('Set up Service Provider X509 Certificiates')
-        install_mellon_cert(options)
-
-        cert_base64 = load_cert_from_file(options.mellon_dst_cert_file)
-        options.sp_signing_cert = cert_base64
-        options.sp_encryption_cert = cert_base64
-
-        logger.step('Build Mellon httpd config file')
-        mellon_httpd_config = build_mellon_httpd_config_file(options)
-        install_file_from_data(mellon_httpd_config,
-                               options.mellon_httpd_config_filename)
-
-        logger.step('Build Mellon SP metadata file')
-        mellon_sp_metadata = build_mellon_sp_metadata_file(options)
-        install_file_from_data(mellon_sp_metadata,
-                               options.mellon_sp_metadata_filename)
-    except Exception as e:
-        if options.show_traceback:
-            traceback.print_exc()
-        print('%s: %s' % (e.__class__.__name__, six.text_type(e)),
-              file=sys.stderr)
-        return STATUS_OPERATION_ERROR
+        try:
+            logger.step('Set up Service Provider X509 Certificiates')
+            install_mellon_cert(options)
+
+            cert_base64 = load_cert_from_file(options.mellon_dst_cert_file)
+            options.sp_signing_cert = cert_base64
+            options.sp_encryption_cert = cert_base64
+
+            logger.step('Build Mellon httpd config file')
+            mellon_httpd_config = build_mellon_httpd_config_file(options)
+            install_file_from_data(mellon_httpd_config,
+                                   options.mellon_httpd_config_filename)
+
+            logger.step('Build Mellon SP metadata file')
+            mellon_sp_metadata = build_mellon_sp_metadata_file(options)
+            install_file_from_data(mellon_sp_metadata,
+                                   options.mellon_sp_metadata_filename)
+        except Exception as e:
+            if options.show_traceback:
+                traceback.print_exc()
+            print('%s: %s' % (e.__class__.__name__, six.text_type(e)),
+                  file=sys.stderr)
+            return STATUS_OPERATION_ERROR
+
+    elif options.module == 'mod_auth_openidc':
+    # ===== Build OIDC metadata =====
+        try:
+            logger.step('Build OIDC SP metadata file')
+            oidc_metadata = build_mellon_sp_metadata_file(options)
```

Comment:
A function called "build_mellon_sp_metadata_file" should not be producing a oidc metadata. Either fix the name or have separate functions, see above comment.